R/new-model.R

```diff
@@ -64,23 +57,18 @@ new_model <- function(
     basename(.path)
   )
 
-  # fill list from passed args
+  # create model object
   .mod <- list()
   .mod[[ABS_MOD_PATH]] <- abs_mod_path
   .mod[[YAML_MOD_TYPE]] <- .model_type
-  if (!is.null(.description)) {
-    checkmate::assert_scalar(.description)
-    .mod[[YAML_DESCRIPTION]] <- .description
-  }
-  if (!is.null(.based_on)) {
-    .mod[[YAML_BASED_ON]] <- safe_based_on(dirname(.path), .based_on)
-  }
-  if (!is.null(.tags)) .mod[[YAML_TAGS]] <- .tags
-  if (!is.null(.bbi_args)) .mod[[YAML_BBI_ARGS]] <- .bbi_args
-
-  # make list into S3 object
   .mod <- create_model_object(.mod, save_yaml = TRUE)
 
+  # update model from passed args
+  if (!is.null(.description)) .mod <- replace_description(.mod, .description)
```

Comment:
Is the plan for these functions (and possibly `modify_model_field()`) to eventually become generics, with methods for NONMEM and possibly other models?

Reply:
I kind of doubt it because the idea of this refactor is that all of these will inherit from `bbi_model` and my current thinking is that all of these `modify_model_field()` helpers should probably _all_ work on any `bbi_model` because they will all have the same metadata structure (in terms of `tags`, `notes`, etc.).

As you note though, this does set us up to make those divergent methods like that if we want to, which is nice.